atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -139,6 +141,15 @@ public TimestampService createTimestampService(
                 initializeAsync);
     }
 
+    @Override
+    public TimestampManagementService createTimestampManagementService(TimestampService timestampService) {
+        Preconditions.checkArgument(timestampService instanceof PersistentTimestampService,
+                "TimestampManagementService must be created based on result of createTimestampService call."
+                        + "\nExpected a PersistentTimestampServiceImpl, got %s", timestampService.getClass());
```

Comment:
nit: the Precondition message should be updated because we're now checking for `PersistentTimestampService`

Also, is there any reason why we check for `PersistentTimestampService` specifically, instead of just the `TimestampManagementService` we want?

Reply:
Hm, that's actually fair, and this could become a default method at that point.

I initially wrote these thinking that different (cassandra/db/etc) factories would have different types for the `TimestampService`, but that proved not to be the case.

Reply:
Done, made the method static and just check if it's castable.